hack/check_pre_req.sh

```diff
@@ -122,7 +122,8 @@ is_installed_kubectl() {
 is_installed_make() {
     is_installed 'GNU Make' \
         'make' \
-        'make --version | head -1'
+        'make --version | head -1' \
+        'sudo apt-get install make or brew install make'
```

Comment:
There are more ways to get `make` on a Mac. I get it from the Xcode Command Line Tools.

Reply:
Understand your point here. The prerequisites specifically call for make to be installed.

A savvy user can use any build tool. A novice user could use the guidance.

Reply:
Shouldn't the novice user on a Mac be told about Xcode Command Line tools?

Reply:
Based on what I am reading, Homebrew installs Xcode CLT on Mac, which should also include `make`. The Mac user should pass this check and not be given the recommendation. My edits to the script came from missing `make` on a Windows WSL build. Noticed the description was missing then.

[https://www.howtogeek.com/211541/install-homebrew-on-mac/](https://www.howtogeek.com/211541/install-homebrew-on-mac/)
[https://mac.install.guide/commandlinetools/8](https://mac.install.guide/commandlinetools/8)

Are you asking for how to install Xcode Command Line Tools?

Reply:
That does not seem quite right to me. I am not asking you to tell me how to install Xcode CLT. I am saying that this script should tell a Mac user (at least) that they can install `make` in at least two ways, by using brew and by using Xcode CLT.

I use a Mac, it has `make` because I installed Xcode CLT, and I have _not_ installed it via Homebrew. The evidence below shows that I have version 3.81 of make, and brew would install version 4.4.1 the license for that is GPL. Apple specifically avoids version 4 and higher of make because it is avoiding GPL. Some corporate developers also avoid GPL.

```
mspreitz@mjs13 ~ % which make
/usr/bin/make

mspreitz@mjs13 ~ % make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0

mspreitz@mjs13 ~ % brew list
(a bunch of stuff, _not_ including make)

mspreitz@mjs13 ~ % brew info make
==> make: stable 4.4.1 (bottled), HEAD
Utility for directing compilation
https://www.gnu.org/software/make/
Not installed
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/m/make.rb
License: GPL-3.0-only
==> Dependencies
Build: lzip ✘
==> Options
--HEAD
	Install HEAD version
==> Caveats
GNU "make" has been installed as "gmake".
If you need to use it as "make", you can add a "gnubin" directory
to your PATH from your bashrc like:

    PATH="/opt/homebrew/opt/make/libexec/gnubin:$PATH"
==> Analytics
install: 11,415 (30 days), 32,712 (90 days), 133,163 (365 days)
install-on-request: 4,522 (30 days), 13,563 (90 days), 56,796 (365 days)
build-error: 11 (30 days)

mspreitz@mjs13 ~ % which gmake
gmake not found
```

Reply:
This could be a simple as changing the commented line to the following.

```
        'sudo apt-get install make or brew install make or, if on a Mac, install Xcode Command Line Tools'
```